src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,134 @@
+package duke;
+
+import duke.command.Command;
+import duke.command.ExitCommand;
+import duke.command.AddCommand;
+import duke.command.DoneCommand;
+import duke.command.DeleteCommand;
+import duke.command.InputCommand;
+import duke.command.GetCommand;
+import duke.command.ListCommand;
+import duke.command.FindCommand;
+
+import java.time.LocalDate;
+
+/**
+ * Represents a parser that parses the input of users.
+ */
+public class Parser {
+
+    /**
+     * Parses the user's input.
+     * @param fullCommand String command by user.
+     * @return Command object.
+     * @throws DukeException If invalid input by user.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+
+        String input = fullCommand;
+        String command = input.split(" ")[0];
+        InputCommand inputCommand;
+
+        try {
+            inputCommand = InputCommand.valueOf(command.toUpperCase());
+        } catch (Exception e) {
+            inputCommand = InputCommand.INVALID;
+        }
+
+        switch (inputCommand) {
+        case BYE:
+            return new ExitCommand();
+        case LIST:
+            return new ListCommand();
+        case DONE:
+            return new DoneCommand(input);
+        case DELETE:
+            return new DeleteCommand(input);
+        case TODO:
+        case DEADLINE:
+        case EVENT:
+            return new AddCommand(input, inputCommand);
+        case GET_TASK:
+            return new GetCommand(input);
+        case FIND:
+            System.out.println("sdfs");
+            return new FindCommand(input);
+        default:
+            throw new DukeException("Give me a valid banana (input)!");
+        }
+
+    }
+
+    /**
+     * Checks if the input has valid index.
+     * @param input String input of user.
+     * @param tasksNumber Number of tasks.
+     * @return true if valid, false if invalid.
+     */
+    public static boolean isValidIndex(String input, int tasksNumber) {
+        String[] stringArray = input.split(" ");
```

Comment:
Perhaps you might want to consider renaming this variable to something like "parsedInput" or "splittedInput" to avoid confusion? (stringArray is a bit too general) 😆 I noticed the same issue in several other places too.